app/src/main/java/app/revanced/integrations/youtube/patches/spoof/requests/StreamingDataRequest.java

```diff
@@ -0,0 +1,190 @@
+        String clientTypeName = clientType.name();
+        Logger.printDebug(() -> "Fetching video streams for: " + videoId + " using client: " + clientType.name());
+
+        try {
+            HttpURLConnection connection = PlayerRoutes.getPlayerResponseConnectionFromRoute(GET_STREAMING_DATA, clientType);
+
+            String authHeader = playerHeaders.get("Authorization");
+            String visitorId = playerHeaders.get("X-Goog-Visitor-Id");
+            connection.setRequestProperty("Authorization", authHeader);
+            connection.setRequestProperty("X-Goog-Visitor-Id", visitorId);
+
+            String innerTubeBody = String.format(PlayerRoutes.createInnertubeBody(clientType), videoId);
+            byte[] requestBody = innerTubeBody.getBytes(StandardCharsets.UTF_8);
+            connection.setFixedLengthStreamingMode(requestBody.length);
+            connection.getOutputStream().write(requestBody);
+
+            final int responseCode = connection.getResponseCode();
+            if (responseCode == 200) return connection;
+
+            handleConnectionError(clientTypeName + " not available with response code: "
+                            + responseCode + " message: " + connection.getResponseMessage(),
+                    null, showErrorToasts);
+        } catch (SocketTimeoutException ex) {
+            handleConnectionError("Connection timeout", ex, showErrorToasts);
+        } catch (IOException ex) {
+            handleConnectionError("Network error", ex, showErrorToasts);
+        } catch (Exception ex) {
+            Logger.printException(() -> "send failed", ex);
+        } finally {
+            Logger.printDebug(() -> "video: " + videoId + " took: " + (System.currentTimeMillis() - startTime) + "ms");
+        }
+
+        return null;
+    }
+
+    private static ByteBuffer fetch(String videoId, Map<String, String> playerHeaders) {
+        final boolean debugEnabled = BaseSettings.DEBUG.get();
+
+        // Retry with different client if empty response body is received.
+        int i = 0;
+        for (ClientType clientType : CLIENT_ORDER_TO_USE) {
+            // Show an error if the last client type fails, or if the debug is enabled then show for all attempts.
+            final boolean showErrorToast = (++i == CLIENT_ORDER_TO_USE.length) || debugEnabled;
+
+            HttpURLConnection connection = send(clientType, videoId, playerHeaders, showErrorToast);
+            if (connection != null) {
+                try {
+                    // gzip encoding doesn't response with content length (-1),
+                    // but empty response body does.
+                    if (connection.getContentLength() != 0) {
+                        try (InputStream inputStream = new BufferedInputStream(connection.getInputStream())) {
+                            try (ByteArrayOutputStream baos = new ByteArrayOutputStream()) {
+                                byte[] buffer = new byte[8192];
```

Comment:
Is the streaming data response really that big? In JSON it's just a small array.